### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'toLowerCase')

### ISSUE THREAD

**Issue by JakeAi** [nativescript-community/ble#268]
iOS 16.3.1

After an xcode update and the cocopod fiasco, I am now unable to use this plugin.

The error is coming from function `CBUUIDToString` when trying to scan.

I startScanning > get result > then I query `device.advertismentData.serviceUUIDs` and look for a specific service. 

```
  TypeError: Cannot read properties of undefined (reading 'toLowerCase')
  at CBUUIDToString (file:///app/vendor.js:1187:26)
  at get serviceUUIDs [as serviceUUIDs] (file:///app/vendor.js:1112:21)
  at file:///app/bundle.js:842:67
  at Array.map (<anonymous>)
```

edit: If I set a breakpoint on the For loop in `get serviceUUIDs()` of `AdvertismentData` it works every time. If I disable the breakpoint, it gives me the error. Which seems like a race condition somewhere?

If I place the breakpoint on `CBUUIDToString`, it is showing the argument being a NSMutableArray. 

If I add the serviceUUIDs get accessor and surround it with a try block, the following line that was failing now works.

```TS
map((d) =>
        d.map((device: Peripheral) => {    
          try {
            device.advertismentData.serviceUUIDs
          } catch (error) {}
          device.isInRecovery = Boolean(device.advertismentData.serviceUUIDs.find((value) => value.toLowerCase() === RecoveryService.uuid.toLowerCase()));
          return device;
        })
      )
    );
```

**Comment by farfromrefug** [comment]
@JakeAi Did you by any change upgrade the ios runtime to 8.5.x ?
If so could you confirm it works with 8.4.x ?

**Comment by JakeAi** [comment]
> @JakeAi Did you by any change upgrade the ios runtime to 8.5.x ? If so could you confirm it works with 8.4.x ?

I am using 8.5.0. However, I am updating my device to 16.4.

**Comment by JakeAi** [comment]
> @JakeAi we are talking @nativescript/ios right? If so please try to use 8.4.x, clean and let me know if it works. I might now what s happening

While I was messing around with the versions, I noticed there was a 8.5.1 for iOS. Would this fix the issue I was having?

https://github.com/NativeScript/ios/commit/b6894346b273b289bfda713a8f0c7055911c945a

**Comment by farfromrefug** [comment]
@JakeAi yes that could really be it!

**Comment by JakeAi** [comment]
> @JakeAi yes that could really be it!

Unreal, that indeed has fixed my issue lol. It wasn't released yet when I updated everything to 8.5.0 lol.
